src/main/java/de/azapps/kafkabackup/common/offset/OffsetSink.java

```diff
@@ -1,71 +1,113 @@
 package de.azapps.kafkabackup.common.offset;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import lombok.RequiredArgsConstructor;
 import org.apache.kafka.clients.admin.AdminClient;
 import org.apache.kafka.clients.admin.ConsumerGroupListing;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.connect.errors.RetriableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @RequiredArgsConstructor
 public abstract class OffsetSink {
 
-  private static final long DEFAULT_SYNC_INTERVAL = 5000;
+    private static final Logger log = LoggerFactory.getLogger(OffsetSink.class);
```

Comment:
cause Lombock is already added, you can just add annotation @Sl4j, and then you don't have to initialize `log` variable

Reply:
Nice! Fixed!